deployment/terraform/output.go

```diff
@@ -3,91 +3,149 @@
+		Value []Instance `json:"value"`
 	} `json:"instances"`
 	DBCluster struct {
-		Value []struct {
-			ClusterEndpoint string `json:"endpoint"`
-			ReaderEndpoint  string `json:"reader_endpoint"`
-		} `json:"value"`
+		Value []DBCluster `json:"value"`
 	} `json:"dbCluster"`
 	Agents struct {
-		Value []struct {
-			PrivateIP  string `json:"private_ip"`
-			PublicIP   string `json:"public_ip"`
-			PublicDNS  string `json:"public_dns"`
-			PrivateDNS string `json:"private_dns"`
-			Tags       struct {
-				Name string `json:"Name"`
-			} `json:"tags"`
-		} `json:"value"`
+		Value []Instance `json:"value"`
 	} `json:"agents"`
 	MetricsServer struct {
-		Value []struct {
-			PrivateIP  string `json:"private_ip"`
-			PublicIP   string `json:"public_ip"`
-			PublicDNS  string `json:"public_dns"`
-			PrivateDNS string `json:"private_dns"`
-		} `json:"value"`
+		Value []Instance `json:"value"`
 	} `json:"metricsServer"`
 	ElasticServer struct {
-		Value []struct {
-			Endpoint string `json:"endpoint"`
-			Tags     struct {
-				Name string `json:"Name"`
-			} `json:"tags"`
-		} `json:"value"`
+		Value []ElasticsearchDomain `json:"value"`
 	} `json:"elasticServer"`
 	S3Bucket struct {
-		Value []struct {
-			Id     string `json:"id"`
-			Region string `json:"region"`
-		}
+		Value []S3Bucket `json:"value"`
 	} `json:"s3Bucket"`
 	S3Key struct {
-		Value []struct {
-			Id     string `json:"id"`
-			Secret string `json:"secret"`
-		}
+		Value []IAMAccess `json:"value"`
 	} `json:"s3Key"`
 }
 
+// Output contains the output variables which are
+// created after a deployment.
+type Output struct {
+	Proxy               Instance            `json:"proxy"`
+	Instances           []Instance          `json:"instances"`
+	DBCluster           DBCluster           `json:"dbCluster"`
+	Agents              []Instance          `json:"agents"`
+	MetricsServer       Instance            `json:"metricsServer"`
+	ElasticSearchServer ElasticsearchDomain `json:"elasticServer"`
+	S3Bucket            S3Bucket            `json:"s3Bucket"`
+	S3Key               IAMAccess           `json:"s3Key"`
+}
+
+type Instance struct {
+	PrivateIP  string `json:"private_ip"`
+	PublicIP   string `json:"public_ip"`
+	PublicDNS  string `json:"public_dns"`
+	PrivateDNS string `json:"private_dns"`
+	Tags       Tags   `json:"tags"`
+}
+
+type ElasticsearchDomain struct {
+	Endpoint string `json:"endpoint"`
+	Tags     Tags   `json:"tags"`
+}
+
+type Tags struct {
+	Name string `json:"Name"`
+}
+
+type DBCluster struct {
+	ClusterEndpoint string `json:"endpoint"`
+	ReaderEndpoint  string `json:"reader_endpoint"`
+}
+
+type IAMAccess struct {
+	Id     string `json:"id"`
+	Secret string `json:"secret"`
+}
+
+type S3Bucket struct {
+	Id     string `json:"id"`
+	Region string `json:"region"`
+}
+
+// Output reads the current terraform output
+func (t *Terraform) Output() (*Output, error) {
+	var buf bytes.Buffer
+	err := t.runCommand(&buf, "output", "-json")
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
```suggestion
	if err := t.runCommand(&buf, "output", "-json"); err != nil {
		return nil, err
	}
```

:police_officer: :stuck_out_tongue:

Reply:
Can we just call this the Claudio coding-style? :laughing: